components/credentials/credentials-form.tsx

```diff
@@ -1,164 +1,150 @@
-					type: "ANONCRED",
-					credentialSchema: schemaId,
-					revocation: formData.revocation
-						? {
-								title: "example_revocation_title",
-								description: "example_revocation_description",
-						  }
-						: null,
-				});
+			setSchemaId(schemaId); 
+			toast.success("Schema created successfully!");
+			const assetPayload: z.infer<typeof AssetRequest> = {
+				mediaType: "image/jpeg",
+				content: credentialLogo ?? "",
+				fileName: "example.jpg",
+				description: "Example asset",
+			};
+			const assetResponse = await createAsset(assetPayload) as typeof AssetResponse._type;
+			const assetId = assetResponse?.asset?.id;
 
-			console.log(
-				":white_check_mark: Credential Definition Created:",
-				credentialDefinitionResponse
-			);
+			const credentialDefinitionPayload: z.infer<
+				typeof CredentialDefinitionRequest
+			> = {
+				name: formData.name || "example_name",
+				version: formData.version || "example_version",
+				credentialSchema: schemaId,
+				identifierType: "DID",
+				identifier: "did:sov:XUeUZauFLeBNofY3NhaZCB",
+				type: "ANONCRED",
+				revocation: formData.revocation
+					? {
+							id: formData.revocation.id,
+							createdAt: formData.revocation.createdAt,
+							updatedAt: formData.revocation.updatedAt,
+							title: formData.revocation.title,
+							description: formData.revocation.description,
+					  }
+					: undefined,
+				icon: assetId,
+			};
 
-			return credentialDefinitionResponse;
+			await createCredentialDefinition(credentialDefinitionPayload);
+			toast.success("Credential definition created successfully!");
+			form.reset();
 		} catch (error) {
-			console.error(
-				":x: Error creating schema, uploading asset, or credential definition:",
-				error
-			);
-			throw error;
+			toast.error("Error creating schema or credential definition.");
+			console.error(error);
+		} finally {
+			setIsSubmitting(false);
 		}
 	};
 
-	const handleCreateCredential = async () => {
+	const handleDeleteCredential = async () => {
+		if (!selectedCredential) return;
+		setIsLoading(true);
 		try {
-			setIsSubmitting(true);
-			setError(null);
-			await createSchemaAndThenDefinition();
-			// Handle success (e.g., reset form, show success message)
+			await deleteCredentialDefinition(selectedCredential.id);
+			setSelectedCredential(null);
+			setIsModalOpen(false);
+			toast.success("Credential deleted successfully!");
 		} catch (error) {
-			console.error("Error during create:", error);
-			setError("There was an error while creating the credential.");
-		} finally {
-			setIsSubmitting(false);
+			setIsModalOpen(false);
+			toast.error("Failed to delete credential.");
+			console.error(error);
 		}
+		setIsLoading(false);
 	};
 
-	useEffect(() => {
-		if (mode === "create") {
-			form.reset();
-		}
-	}, [mode, form]);
-
-	useEffect(() => {
-		if (mode === "create" && issuers.length === 0) {
-			fetchIssuers();
-		}
-	}, [mode, issuers, fetchIssuers]);
+	const onSubmit = () => {
```

Comment:
```suggestion
	const onSubmit = (data: CredentialDefinitionRequest) => { 
```
onSubmit of the form, pass the data for you here, you can handle the data here, once  you don't need to use like 
```
const formData: any = form.getValues();
```